tests/rptest/tests/partition_balancer_test.py

```diff
@@ -468,32 +468,27 @@ def test_full_nodes(self):
             },
             environment={"__REDPANDA_TEST_DISK_SIZE": disk_size})
 
-        def get_disk_usage(node):
-            for line in node.account.ssh_capture(
-                    "df --block-size 1 /var/lib/redpanda"):
-                self.logger.debug(line.strip())
-                if '/var/lib/redpanda' in line:
-                    return int(line.split()[2])
-            assert False, "couldn't parse df output"
-
-        def get_total_disk_usage():
-            return sum(get_disk_usage(n) for n in self.redpanda.nodes)
-
         self.topic = TopicSpec(partition_count=32)
         self.client().create_topic(self.topic)
 
         # produce around 2GB of data, this should be enough to fill node disks
         # to a bit more than 70% usage on average
+        msg_count = 19_000
         producer = KgoVerifierProducer(self.test_context,
                                        self.redpanda,
                                        self.topic,
                                        msg_size=102_400,
-                                       msg_count=19_000)
+                                       msg_count=msg_count)
         producer.start(clean=False)
+        producer.wait_for_acks(msg_count, timeout_sec=120, backoff_sec=5)
 
-        wait_until(lambda: get_total_disk_usage() / 5 / disk_size > 0.7,
-                   timeout_sec=120,
-                   backoff_sec=5)
+        def get_disk_usage(node):
```

Comment:
nit: can we make this a method on `Redpanda` service ?

Reply:
good idea